include/itkWaveletCoeffsInverseFFTImageFilter.hxx

```diff
@@ -0,0 +1,133 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkWaveletCoeffsInverseFFTImageFilter_hxx
+#define itkWaveletCoeffsInverseFFTImageFilter_hxx
+#include "itkWaveletCoeffsInverseFFTImageFilter.h"
+#include "itkForwardFFTImageFilter.h"
+#include "itkInverseFFTImageFilter.h"
+#include "itkFFTPadImageFilter.h"
+#include "itkWaveletFrequencyForward.h"
+#include "itkWaveletFrequencyInverse.h"
+#include "itkWaveletFrequencyForwardUndecimated.h"
+#include "itkWaveletFrequencyInverseUndecimated.h"
+#include "itkWaveletFrequencyFilterBankGenerator.h"
+#include "itkHeldIsotropicWavelet.h"
+#include "itkVowIsotropicWavelet.h"
+#include "itkSimoncelliIsotropicWavelet.h"
+#include "itkShannonIsotropicWavelet.h"
+#include "itkMonogenicSignalFrequencyImageFilter.h"
+#include "itkVectorInverseFFTImageFilter.h"
+#include "itkPhaseAnalysisSoftThresholdImageFilter.h"
+#include "itkZeroDCImageFilter.h"
+#include "itkImage.h"
+#include "itkCastImageFilter.h"
+#include "itkNumberToString.h"
+#include <string>
+
+
+namespace itk
+{
+template< typename TImageType, 
+  typename TWaveletFunction > 
+WaveletCoeffsInverseFFTImageFilter< TImageType, TWaveletFunction > 
+:: WaveletCoeffsInverseFFTImageFilter()
+{
+  m_Levels = 4;
+  m_HighPassSubBands = 3;
+}
+
+template< typename TImageType, 
+  typename TWaveletFunction > 
+void 
+WaveletCoeffsInverseFFTImageFilter< TImageType, TWaveletFunction > 
+:: GenerateData()
+{
+  m_FFTPadFilter = FFTPadType::New();
+  m_ZeroDCFilter = ZeroDCType::New();
+  m_ForwardFFTFilter = FFTForwardType::New();
+  m_ForwardWaveletFilter = ForwardWaveletType::New();
```

Comment:
Since this is only required once per filter instance, these should be in the constructor.